includes/sensei-functions.php

```diff
@@ -281,14 +281,42 @@ function sensei_user_login_url() {
  * duplicate of Sensei()->access_settings().
  *
  * @since 1.9.0
+ * @since 3.5.2 Added hook to filter the return.
  * @return bool
  */
 function sensei_is_login_required() {
+	global $post;
 
-	$login_required = isset( Sensei()->settings->settings['access_permission'] ) && ( true == Sensei()->settings->settings['access_permission'] );
+	$post_type = get_post_type( $post );
+	$course_id = null;
 
-	return $login_required;
+	if ( 'course' === $post_type ) {
+		$course_id = $post->ID;
+	} elseif ( 'lesson' === $post_type ) {
+		$course_id = Sensei()->lesson->get_course_id( $post->ID );
+	}
 
+	if (
+		! $course_id
+		|| 'course' !== get_post_type( $course_id )
```

Comment:
Why do we need this check?

Reply:
Hum, the second one we really didn't need. It's being checked here: `get_course_id`. Updated here: https://github.com/Automattic/sensei/commit/2bb7fb558b3d3129be5183c1d3ea0e565127b11e
The `! $course_id` check I kept. Although the `get_course_id` returns `false` for invalid cases, I think here makes more sense use `null`, do you agree?